source/entities/Creature.cpp

```diff
@@ -2967,6 +3040,38 @@ void Creature::refreshCreature(ODPacket& packet)
     OD_ASSERT_TRUE(packet >> mLevel);
     OD_ASSERT_TRUE(packet >> mOverlayHealthValue);
     RenderManager::getSingleton().rrScaleEntity(this);
+
+    uint32_t nbEffects;
+    OD_ASSERT_TRUE(packet >> nbEffects);
+
+    // We copy the list of effects currently on this creature. That will allow to
+    // apply only effects we were not allowed of
```

Comment:
I meant "we are not aware of" (ie since all effects are sent when refreshing, we only create the ones that are not already created on client side - to avoid having to know for each client if the effect is known or not)